go/runtime/client/api/grpc.go

```diff
@@ -0,0 +1,496 @@
+	dec func(interface{}) error,
+	interceptor grpc.UnaryServerInterceptor,
+) (interface{}, error) {
+	var rq QueryTxsRequest
+	if err := dec(&rq); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return srv.(RuntimeClient).QueryTxs(ctx, &rq)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: methodQueryTxs.Full(),
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return srv.(RuntimeClient).QueryTxs(ctx, req.(*QueryTxsRequest))
+	}
+	return interceptor(ctx, &rq, info, handler)
+}
+
+func handlerWaitBlockIndexed( // nolint: golint
+	srv interface{},
+	ctx context.Context,
+	dec func(interface{}) error,
+	interceptor grpc.UnaryServerInterceptor,
+) (interface{}, error) {
+	var rq WaitBlockIndexedRequest
+	if err := dec(&rq); err != nil {
+		return nil, err
+	}
+	if interceptor == nil {
+		return nil, srv.(RuntimeClient).WaitBlockIndexed(ctx, &rq)
+	}
+	info := &grpc.UnaryServerInfo{
+		Server:     srv,
+		FullMethod: methodWaitBlockIndexed.Full(),
+	}
+	handler := func(ctx context.Context, req interface{}) (interface{}, error) {
+		return nil, srv.(RuntimeClient).WaitBlockIndexed(ctx, req.(*WaitBlockIndexedRequest))
+	}
+	return interceptor(ctx, &rq, info, handler)
+}
+
+func handlerWatchBlocks(srv interface{}, stream grpc.ServerStream) error {
+	var runtimeID signature.PublicKey
+	if err := stream.RecvMsg(&runtimeID); err != nil {
+		return err
+	}
+
+	ctx := stream.Context()
+	ch, sub, err := srv.(RuntimeClient).WatchBlocks(ctx, runtimeID)
+	if err != nil {
+		return err
+	}
+	defer sub.Close()
+
+Loop:
+	for {
+		select {
+		case blk, ok := <-ch:
+			if !ok {
+				break Loop
+			}
+
+			if err := stream.SendMsg(blk); err != nil {
+				return err
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+
+	return nil
+}
+
+// RegisterService registers a new runtime client service with the given gRPC server.
+func RegisterService(server *grpc.Server, service RuntimeClient) {
+	server.RegisterService(&serviceDesc, service)
+}
+
+type runtimeClient struct {
+	enclaverpc.Transport
+
+	conn *grpc.ClientConn
+}
+
+func (c *runtimeClient) SubmitTx(ctx context.Context, request *SubmitTxRequest) ([]byte, error) {
+	var rsp []byte
+	if err := c.conn.Invoke(ctx, methodSubmitTx.Full(), request, &rsp); err != nil {
+		return nil, err
+	}
+	return rsp, nil
+}
+
+func (c *runtimeClient) GetBlock(ctx context.Context, request *GetBlockRequest) (*block.Block, error) {
+	var rsp block.Block
+	if err := c.conn.Invoke(ctx, methodGetBlock.Full(), request, &rsp); err != nil {
+		return nil, err
+	}
+	return &rsp, nil
+}
+
+func (c *runtimeClient) GetBlockByHash(ctx context.Context, request *GetBlockByHashRequest) (*block.Block, error) {
+	var rsp block.Block
+	if err := c.conn.Invoke(ctx, methodGetBlockByHash.Full(), request, &rsp); err != nil {
+		return nil, err
+	}
+	return &rsp, nil
+}
+
+func (c *runtimeClient) GetTx(ctx context.Context, request *GetTxRequest) (*TxResult, error) {
```

Comment:
Same question as for the rust code - should we use `Tx` or `Txn` for transaction?